File: python/1762. Buildings With an Ocean View/solution.py
class Solution:
    def findBuildings(self, heights: List[int]) -> List[int]:
        answer = []
        currentMax = 0

        for i in range(len(heights) -1, -1, -1):
            if heights[i] > currentMax:
                answer.append(i)
            currentMax = max(currentMax, heights[i])

        return reversed(answer)

File: java/1762. Buildings With an Ocean View/Solution.java
class Solution {
    public int[] findBuildings(int[] heights) {
        List<Integer> buildings = new ArrayList<>();
        int currentMax = 0;

        for (int i = heights.length - 1; i >= 0; i--) {
            if(heights[i] > currentMax) buildings.add(i);
            currentMax = Math.max(currentMax, heights[i]);
        }

        int[] answer = new int[buildings.size()];
        for (int i = 0; i < buildings.size(); i++) {
            answer[i] = buildings.get(buildings.size() - 1 -i);
        }

        return answer;

    }
}